internal/buffer/settings.go

```diff
@@ -117,6 +120,15 @@ func (b *Buffer) DoSetOptionNative(option string, nativeValue interface{}) {
 	if b.OptionCallback != nil {
 		b.OptionCallback(option, nativeValue)
 	}
+
+	_, err := config.RunPluginFnBool(b.Settings, "onBufferOptionChanged",
+		luar.New(ulua.L, b),
+		luar.New(ulua.L, option),
+		luar.New(ulua.L, string(oldValue.(string))),
```

Comment:
This is what happens when changing any non-string option:

```
Micro encountered an error: *runtime.TypeAssertionError interface conversion: interface {} is bool, not string
runtime/iface.go:261 (0x40a575)
github.com/zyedidia/micro/v2/internal/buffer/settings.go:127 (0x839adb)
github.com/zyedidia/micro/v2/internal/action/command.go:656 (0x87e9df)
github.com/zyedidia/micro/v2/internal/action/command.go:673 (0x87ec59)
github.com/zyedidia/micro/v2/internal/action/command.go:703 (0x87ef94)
github.com/zyedidia/micro/v2/internal/action/command.go:1140 (0x882308)
github.com/zyedidia/micro/v2/internal/action/actions.go:1859 (0x870d72)
github.com/zyedidia/micro/v2/internal/info/infobuffer.go:163 (0x840463)
github.com/zyedidia/micro/v2/internal/action/infopane.go:223 (0x886d2c)
github.com/zyedidia/micro/v2/internal/action/infopane.go:54 (0x885c36)
github.com/zyedidia/micro/v2/internal/action/infopane.go:132 (0x8865a2)
github.com/zyedidia/micro/v2/internal/action/infopane.go:91 (0x88616c)
github.com/zyedidia/micro/v2/cmd/micro/micro.go:482 (0x8c4425)
github.com/zyedidia/micro/v2/cmd/micro/micro.go:400 (0x8c3e3f)
runtime/proc.go:255 (0x436ca7)
runtime/asm_amd64.s:1581 (0x465de1)
 
If you can reproduce this error, please report it at https://github.com/zyedidia/micro/issues
```

Why are you even converting it to string?

Reply:
Damn, intentionally when this PR was created...but long before my deep dive into `micro`'s config environment. :man_facepalming: